### DISCUSSION TITLE
How to hide api-key and secret that is passed as part of headers?

### DISCUSSION THREAD

**Discussion by satyamtpatel** [grafana/grafana-infinity-datasource#177] [Q&A] [answered]
When we are accessing our API, we are passing api-key as part of headers. When we perform inspect in grafana, user can see those api-key using inspect panel. How we can hide it?

**Comment by yesoreyeram** [comment, marked as answer]
Use the api keys in the datasource configuration. Strongly don't use api keys in the panel or query.

**Comment by satyamtpatel** [comment]
1) So if we want to use multiple APIs then how we can setup?
2) Isn’t url under data source is deprecated?

Sent from my iPhone

On Oct 23, 2021, at 3:11 PM, Sriram ***@***.***> wrote:

﻿
EXTERNAL SENDER: use caution with links and attachments.

Use the api keys in the datasource configuration. Strongly don't use api keys in the panel or query.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fyesoreyeram%2Fgrafana-infinity-datasource%2Fdiscussions%2F177%23discussioncomment-1525943&data=04%7C01%7Cspatel%40kyrasolutions.com%7C4a28553f36974f9a537108d99658dce9%7C1c74c8556fc54416824e7ee0d3967823%7C1%7C0%7C637706130683357704%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=9jLNUK%2B3FEUaWsxQeG8FXjogVu5hguEyoKjXbEdRwJY%3D&reserved=0>, or unsubscribe<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAH77XDEFV2LTLBROXFECD3TUIMCERANCNFSM5GSQ7FQQ&data=04%7C01%7Cspatel%40kyrasolutions.com%7C4a28553f36974f9a537108d99658dce9%7C1c74c8556fc54416824e7ee0d3967823%7C1%7C0%7C637706130683357704%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=y%2F4fXwL9Fb%2BoJ5r%2Fy%2BqutTBAe1BVv0r3Ljo9sKdGayA%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Cspatel%40kyrasolutions.com%7C4a28553f36974f9a537108d99658dce9%7C1c74c8556fc54416824e7ee0d3967823%7C1%7C0%7C637706130683367698%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=dxqV%2FKNruVk495a%2BqyPhkj8RGZ0xOjtUlq%2FqnwlpmHY%3D&reserved=0> or Android<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Cspatel%40kyrasolutions.com%7C4a28553f36974f9a537108d99658dce9%7C1c74c8556fc54416824e7ee0d3967823%7C1%7C0%7C637706130683367698%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=Tpje1WQSJtzgo5vsfm1Xw7E7UBAFPDtLKhqiuU803aA%3D&reserved=0>.

Disclaimer: Kyra Solutions Inc, EMAIL CONFIDENTIALITY NOTICE: This email is intended only for the use of the individual or entity to which it is addressed and may contain information that is privileged, confidential and exempt from disclosure under applicable law. If you have received this communication in error, please do not distribute, but delete the original message. Please notify the sender by email at the address shown and delete it completely from your computer system. Thank you for your compliance.

**Reply by yesoreyeram** [reply]
1. If you want to use multiple APIs with different API keys, create one datasource per instance/domain name.
2. yes URL under datasource is deprecated. But the other options like headers are valid.

**Comment by satyamtpatel** [comment]
1) You mean we have to use separate grafana for each api. Sorry, I want to understand what you mean per instance/domain name. Thanks

Sent from my iPhone

On Oct 23, 2021, at 5:14 PM, Sriram ***@***.***> wrote:

﻿
EXTERNAL SENDER: use caution with links and attachments.

  1.  If you want to use multiple APIs with different API keys, create one datasource per instance/domain name.
  2.  yes URL under datasource is deprecated. But the other options like headers are valid.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fyesoreyeram%2Fgrafana-infinity-datasource%2Fdiscussions%2F177%23discussioncomment-1526158&data=04%7C01%7Cspatel%40kyrasolutions.com%7C6509a4cba49c494efd3808d9966a20a7%7C1c74c8556fc54416824e7ee0d3967823%7C1%7C0%7C637706204819272276%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=4A9NHCLm08zIyLyiYyu0R2P%2BGw%2F%2B3vxGnz0avWg678Y%3D&reserved=0>, or unsubscribe<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAH77XDGA7ODSIYWT2C4MHKDUIMQT5ANCNFSM5GSQ7FQQ&data=04%7C01%7Cspatel%40kyrasolutions.com%7C6509a4cba49c494efd3808d9966a20a7%7C1c74c8556fc54416824e7ee0d3967823%7C1%7C0%7C637706204819272276%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=Mun5ojOMsWunNROW0asRWXhQvQtYDHnz%2BDsjTJEciZY%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Cspatel%40kyrasolutions.com%7C6509a4cba49c494efd3808d9966a20a7%7C1c74c8556fc54416824e7ee0d3967823%7C1%7C0%7C637706204819282273%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=UgrWm9GFGPbGBVd22%2BclJdCZrr6jLZl2RPxjn%2BBtZxs%3D&reserved=0> or Android<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Cspatel%40kyrasolutions.com%7C6509a4cba49c494efd3808d9966a20a7%7C1c74c8556fc54416824e7ee0d3967823%7C1%7C0%7C637706204819282273%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ukN2nF3kHRTfgUzIPeWIg8HZpdACIS%2B6CMUXlODsL0o%3D&reserved=0>.

Disclaimer: Kyra Solutions Inc, EMAIL CONFIDENTIALITY NOTICE: This email is intended only for the use of the individual or entity to which it is addressed and may contain information that is privileged, confidential and exempt from disclosure under applicable law. If you have received this communication in error, please do not distribute, but delete the original message. Please notify the sender by email at the address shown and delete it completely from your computer system. Thank you for your compliance.

**Reply by yesoreyeram** [reply]
In grafana datasources page, create individual datasource for each API.
